plotly/offline.py

```diff
@@ -0,0 +1,178 @@
+""" Plotly Offline
+    A module to use Plotly's graphing library with Python
+    without connecting to a public or private plotly enterprise
+    server.
+"""
+from __future__ import absolute_import
+
+import uuid
+import json
+import os
+import requests
+
+import plotly.plotly as py
+from plotly import utils
+from plotly.exceptions import PlotlyError
+
+PLOTLY_OFFLINE_DIRECTORY = plotlyjs_path = os.path.expanduser(
+    os.path.join(*'~/.plotly/plotlyjs'.split('/')))
+PLOTLY_OFFLINE_BUNDLE = os.path.join(PLOTLY_OFFLINE_DIRECTORY,
+                                     'plotly-ipython-offline-bundle.js')
+
+__PLOTLY_OFFLINE_INITIALIZED = False
+
+
+def download_plotlyjs(download_url):
+    if not os.path.exists(plotlyjs_path):
+        os.makedirs(plotlyjs_path)
+
+    res = requests.get(download_url)
+    res.raise_for_status()
+
+    with open(PLOTLY_OFFLINE_BUNDLE, 'wb') as f:
+        f.write(res.content)
+
+    print('\n'.join([
+        'Success! Now start an IPython notebook and run the following ' +
+        'code to make your first offline graph:',
+        '',
+        'import plotly',
+        'plotly.offline.init_notebook_mode() '
+        '# run at the start of every ipython notebook',
+        'plotly.offline.iplot([{"x": [1, 2, 3], "y": [3, 1, 6]}])'
+    ]))
+
+
+def init_notebook_mode():
```

Comment:
is this function necessary (non-blocking for now, but just out of curiosity)? since `iplot` is really `offline.iplot`, could we just run this as needed?